### ISSUE TITLE
Folding not saved for every file

### ISSUE THREAD

**Issue by setalosas** [forivall/atom-remember-folds#27]
If I have open lots of files (30), the foldings are saved only with certain files, not with others.

I found these messages in the console:

![image](https://user-images.githubusercontent.com/21111643/30770236-6e3b8df0-a02b-11e7-9623-124895585602.png)

So I looked into the source of the messages:

```
  saveEditorFolds(editor, buffer) {
    const displayLayer = buffer.getDisplayLayer(0);
    if (displayLayer == null) {
      console.error('Remember Folds could not get display layer', buffer.displayLayers)
      return
    }

    const uri = editor.getURI()
    if (uri == null) {
      debug('Remember folds ignoring text editor without uri')
      return
    }

    const foldsByMarkerId = displayLayer.foldsMarkerLayer.markersById
    const folds = []
    for (const markerId in foldsByMarkerId) {
      const fold = foldsByMarkerId[markerId]
      folds.push(fold.getRange())
    }
    this.folds[uri] = folds
    this.lineCounts[uri] = editor.getLineCount()
  }
```
It seems that the code assumes that displayLayer[0] - whatever is it - always exists, but it's not the case:

![image](https://user-images.githubusercontent.com/21111643/30770284-9a5a3854-a02c-11e7-82eb-c70ba96a9293.png)

The data structure shows that there is only displayLayer[1]:

![image](https://user-images.githubusercontent.com/21111643/30770287-af243f64-a02c-11e7-93b6-d385ab9e1354.png)

There ares file with displayLayers 6 and 7 too, as you can see from the error messages above. So I tried to fix it with a loop instead of the fix 0 index:

```
saveEditorFolds(editor, buffer) {
  for (const dlKey in buffer.displayLayers) {
	var displayLayer
    if(buffer.displayLayers.hasOwnProperty(dlKey) && (displayLayer = buffer.displayLayers[dlKey])) {
      const uri = editor.getURI()
      if (uri === null) {
        debug('Remember folds ignoring text editor without uri')
        return
      }
      const foldsByMarkerId = displayLayer.foldsMarkerLayer.markersById
      const folds = []
      for (const markerId in foldsByMarkerId) {
        const fold = foldsByMarkerId[markerId]
        folds.push(fold.getRange())
      }
      this.folds[uri] = folds
      this.lineCounts[uri] = editor.getLineCount()
      return
    }
  } 
  console.error('Remember Folds could not get display layer', buffer.displayLayers)
}
```

And it works! It's a bit ugly that I access directly the property instead of calling the method, but in the original code there is a similar access in the error message. Icould use the method, but I have no idea what range should I check - and that's slower too.

As I don't know what is a displayLayer, I have no idea, whether this bug comes from my setup or not. (Sorry, I have no idea how to include a package list, but I can lookup another Atom issue I reported with full info before if needed.)

**Comment by forivall** [comment]
Display layers (iirc) are the things that contain the folding markers, the linter error markers, etc.

Thanks! I'll try to take a look into this soon, so i can figure out a way of detecting which display layer is the folding layer.

You can generate a list of your plugins with the `apm list` command in the terminal.

**Comment by the-question** [comment]
Any news on this?
Same error here (Remember Folds could not get display layer).
Config is: Fedora 27, Atom 1.22.1-0.1 (installed from RPM on atom.io), remember-folds 0.3.0.

I can post my `apm list` if needed.

**Comment by forivall** [comment]
No, I don't really use atom at all anymore, and i'm not interested in maintaining this unless i have a reason to go back to atom.
